util/http_client_pool.go

```diff
@@ -37,7 +41,30 @@ func init() {
 }
 
 func GetHttpClient() *http.Client { return httpClientPool.Get().(*http.Client) }
+func GetHttpClientForOauth2(oauthToken string) (*http.Client, error) {
+	if oauthToken != "" {
```

Comment:
should we error if it's empty?

Reply:
In this case, yes, because the oauth2 transport should not be used for empty tokens. Fixed